base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -461,7 +488,13 @@ where B: BlockchainBackend + 'static
             source_peer
         );
 
-        let block = self.reconcile_block(source_peer.clone(), new_block).await?;
+        let block_result = self.reconcile_block(source_peer.clone(), new_block).await;
+
+        {
```

Comment:
If we allow the block before adding it, the block could reconcile again. Any reason not to unlock after handle_block?